noir/noir-repo/compiler/noirc_evaluator/src/ssa/ssa_gen/mod.rs

```diff
@@ -701,24 +701,29 @@ impl<'a> FunctionContext<'a> {
         assert_message: &Option<Box<(Expression, HirType)>>,
     ) -> Result<Option<ConstrainError>, RuntimeError> {
         let Some(assert_message_payload) = assert_message else { return Ok(None) };
-
-        if let Expression::Literal(ast::Literal::Str(static_string)) = &assert_message_payload.0 {
-            return Ok(Some(ConstrainError::StaticString(static_string.clone())));
-        }
-
         let (assert_message_expression, assert_message_typ) = assert_message_payload.as_ref();
 
-        let values = self.codegen_expression(assert_message_expression)?.into_value_list(self);
+        Ok(Some(
+            if let Expression::Literal(ast::Literal::Str(static_string)) = assert_message_expression
```

Comment:
```suggestion
        let error = if let Expression::Literal(ast::Literal::Str(static_string)) = assert_message_expression {
           ....
         };
         
         Ok(Some(error))
        
```

Can't give a full suggestion but I think it would be best to seperate the wrappers from the if-else.